File: data/src/main/java/com/github/anrimian/musicplayer/data/database/converters/DateConverter.java
package com.github.anrimian.musicplayer.data.database.converters;

import androidx.room.TypeConverter;

import java.util.Date;

public class DateConverter {

    @TypeConverter
    public Long toMillis(Date date) {
        if (date == null) {
            return null;
        } else {
            return date.getTime();
        }
    }

    @TypeConverter
    public Date toDate(Long millis) {
        if (millis == null) {
            return null;
        } else {
            return new Date(millis);
        }
    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/database/converters/DateConverter.kt
package com.github.anrimian.musicplayer.data.database.converters

import androidx.room.TypeConverter
import java.util.Date

class DateConverter {

    @TypeConverter
    fun toMillis(date: Date?): Long? = date?.time

    @TypeConverter
    fun toDate(millis: Long?): Date? = if (millis == null) { null } else { Date(millis) }

}
